cloud/scope/powervs_cluster.go

```diff
@@ -1647,24 +1669,19 @@ func (s *PowerVSClusterScope) ReconcileTransitGateway() (bool, error) {
 
 	// check the status and update the transit gateway's connections if they are not proper
 	if tg != nil {
-		requeue, powerVSConn, vpcConn, err := s.checkAndUpdateTransitGateway(tg, true)
+		requeue, err := s.checkAndUpdateTransitGateway(tg)
 		if err != nil {
 			return false, err
 		}
-		s.SetTransitGatewayStatus(tg.ID, ptr.To(false), powerVSConn, vpcConn)
```

Comment:
But while creating I am setting(need to) the status immediately after create call in s.createTransitGateway(), so I feel its ok to keep inside isTransitGatewayExists() because that func meant to process the existing resource, so when we set controllercreated to false, it would be good if we keep it inside that func.

> Rather than embedding both inside checkAndUpdateTransitGateway?

Then the main Reconcile would be too long if I add these three calls and check the error, it's better to keep them in this func IMO.